interfaces/cython/cantera/composite.py

```diff
@@ -631,14 +633,15 @@ def restore_data(self, data, labels):
         full_states = self._phase._full_states.values()
         if isinstance(self._phase, PureFluid):
             # ensure that potentially non-unique state definitions are checked last
-            last = ['TP', 'TX', 'PX']
+            last = ['TP', 'TQ', 'PQ']
             full_states = [fs for fs in full_states
-                           if fs not in last] + ['TPX'] + last
+                           if fs not in last] + ['TPQ'] + last
 
         # determine whether complete concentration is available (mass or mole)
-        # assumes that `X` or `Y` is always in last place
+        # assumes that 'X' or 'Y' is always in last place
+        # (for PureFluid, 'Q_H2O' is treated in a similar fashion)
         mode = ''
-        for prefix in ['X_', 'Y_']:
+        for prefix in ['X_', 'Y_', 'Q_']:
```

Comment:
With this PR, the mole fraction `X_H2O` will become accessible again (instead of being masked by the quality. For consistency, I'd prefer to keep things as is (it's also much easier to follow the implementation).

Reply:
see:
https://github.com/Cantera/cantera/blob/54c12fc59cfd10a6da2bb7cb81980cd1f7a24696/interfaces/cython/cantera/composite.py#L796-L797

Reply:
I think if you treat `Q` like another scalar variable (still unique to the `PureFluid` class, of course), rather than an "n_species" variable, this should work out. I don't have any problem with the per-species variables including `X` and `Y` still ending up like `X_speciesName`.

Reply:
I believe there is a future use case for multi-component fluids (i.e. *not* pure fluids), that will have dissimilar vapor pressures (and thus different `Q` values). From that perspective, I'd suggest to keep the species name attached and treat vapor fraction as an `_nspecies` variable.

Reply:
resolved ... made `Q` a scalar